packages/blitz-dom/src/document.rs

```diff
@@ -222,6 +227,12 @@ impl DocumentLike for Document {
                             }
                             self.set_focus_to(node_id);
                         }
+                    } else if el.name.local == local_name!("a") {
+                        if let Some(href) = el.attr(local_name!("href")) {
+                            self.navigation_provider.navigate_new_page(href.to_owned());
```

Comment:
We need to call `Document::resolve_url` (or equivalent functionality) on the href before sending it to the navigation provider in order to support relative links. I ran into borrow checker issues doing this. One solution might be to clone the base_url earlier in this method and move the resolution logic to a free functions (or wrap the base_url in a newtype with the method on it).